synapse/storage/event_push_actions.py

```diff
@@ -353,12 +362,14 @@ def f(txn):
                     before_clause += " "
                 before_clause += "AND epa.highlight = 1"
```

Comment:
Given the shape of the index it might be better to run the query twice with ``highlight = 1`` and ``highlight = 0`` when querying for notifications when only_highlight is False.